alf/layers.py

```diff
@@ -3612,3 +3612,57 @@ def forward(self, input):
                 lambda x: x.float() if x.dtype.is_floating_point else x, input)
         with torch.cuda.amp.autocast(self._enabled):
             return self._net(input)
+
+
+class PeriodicReset(torch.nn.Module):
```

Comment:
Is it possible to use  `common.Periodically`?

Reply:
`common.Periodically` maintains its own counter, which is slightly different than this.